services/command-router/src/main/java/org/eclipse/hono/commandrouter/ApplicationConfig.java

```diff
@@ -172,13 +174,28 @@ public ObjectFactoryCreatingFactoryBean amqpServerFactory() {
     /**
      * Creates a new instance of an AMQP 1.0 protocol handler for Hono's <em>Command Router</em> API.
      *
-     * @param service The service instance to delegate to.
      * @return The handler.
      */
     @Bean
     @Scope("prototype")
-    public AmqpEndpoint commandRouterAmqpEndpoint(final CommandRouterService service) {
-        return new DelegatingCommandRouterAmqpEndpoint<>(vertx(), service);
+    public AmqpEndpoint commandRouterAmqpEndpoint() {
+        final CommandRouterService commandRouterService = commandRouterService();
+        return new DelegatingCommandRouterAmqpEndpoint<>(vertx(), commandRouterService) {
+
+            @Override
+            public void registerLivenessChecks(final HealthCheckHandler handler) {
```

Comment:
how about adding this as the default implementation to `org.eclipse.hono.service.amqp.AbstractDelegatingRequestResponseEndpoint`?
It seems to make a lot of sense to register these checks automatically, doesn't it?

Reply:
I also thought about this. But that would have implications on the `hono-service-device-connection` module and its usage of the `DelegatingDeviceConnectionAmqpEndpoint` class, where the given service implementation is already registered with the health check by other means (done in the `Application` class there).

Reply:
I see, then let's keep it like this for the time being ...